lib/deps.js

```diff
@@ -0,0 +1,78 @@
+'use strict';
+
+function makeErrorModule(error) {
+  return new Proxy(
+    {},
+    {
+      get: () => {
+        throw error;
+      },
+      set: () => {
+        throw error;
+      }
+    }
+  );
+}
+
+function retrieveKerberos() {
+  try {
+    return require('kerberos');
+  } catch (err) {
+    const noKerberosError = new Error(
+      'The `kerberos` module was not found. Please install it and try again.'
+    );
+    return makeErrorModule(noKerberosError);
+  }
+}
+
+// Facilitate loading EJSON optionally
+function retrieveEJSON() {
+  try {
+    return require('mongodb-extjson');
+  } catch (error) {} // eslint-disable-line
+
+  // Throw an error if an attempt to use EJSON is made when it is not installed
+  const noEJSONError = new Error(
+    'The `mongodb-extjson` module was not found. Please install it and try again.'
+  );
+  return makeErrorModule(noEJSONError);
+}
+
+function retrieveBSON() {
+  const BSON = require('bson');
+  BSON.native = false;
```

Comment:
let's look into whether this `native` tag is relevant, I'd prefer to drop it if not